src/services/BadgesService.ts

```diff
@@ -269,4 +270,19 @@ export class BadgesService {
       recipients.map((recipient) => recipient.address)
     )
   }
+
+  static giveFinishProposalBadges(proposalsWithOutcome: ProposalWithOutcome[]) {
+    const passedProposals = proposalsWithOutcome.filter((proposal) => proposal.newStatus === ProposalStatus.Passed)
+    inBackground(async () => {
+      try {
+        await BadgesService.giveLegislatorBadges(passedProposals)
+      } catch (error) {
+        ErrorService.report('Error while attempting to give badges', {
+          error,
+          category: ErrorCategory.Badges,
+          passedGovernanceProposals: passedProposals,
```

Comment:
`passedProposals` includes all proposal types here as the filtering is done in `giveLegislatorBadges`. You could rename `passedGovernanceProposals` to `passedProposals` or you could move the governance type filter to this function and remove it from the `giveLegislatorBadges`.